x-pack/plugins/security_solution/common/endpoint/data_generators/endpoint_action_generator.ts

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { DeepPartial } from 'utility-types';
+import { merge } from 'lodash';
+import { BaseDataGenerator } from './base_data_generator';
+import { EndpointActionData, ISOLATION_ACTIONS } from '../types';
+
+interface EcsError {
+  code: string;
+  id: string;
+  message: string;
+  stack_trace: string;
+  type: string;
+}
+
+interface EndpointActionFields {
+  action_id: string;
+  data: EndpointActionData;
+}
+
+interface ActionRequestFields {
+  expiration: string;
+  type: 'INPUT_ACTION';
+  input_type: 'endpoint';
+}
+
+interface ActionResponseFields {
+  completed_at: string;
+  started_at: string;
+}
+export interface LogsEndpointAction {
+  '@timestamp': string;
+  agent: {
+    id: string | string[];
+  };
+  EndpointAction: EndpointActionFields & ActionRequestFields;
+  error?: EcsError;
+  user: {
+    id: string;
+  };
+}
+
+export interface LogsEndpointActionResponse {
+  '@timestamp': string;
+  agent: {
+    id: string | string[];
+  };
+  EndpointAction: EndpointActionFields & ActionResponseFields;
+  error?: EcsError;
+}
+
+const ISOLATION_COMMANDS: ISOLATION_ACTIONS[] = ['isolate', 'unisolate'];
+
+export class EndpointActionGenerator extends BaseDataGenerator {
+  /** Generate a random endpoint Action request (isolate or unisolate) */
+  generate(overrides: DeepPartial<LogsEndpointAction> = {}): LogsEndpointAction {
+    const timeStamp = new Date(this.randomPastDate());
+    return merge(
+      {
+        '@timestamp': timeStamp.toISOString(),
+        agent: {
+          id: [this.randomUUID()],
+        },
+        EndpointAction: {
+          action_id: this.randomUUID(),
+          expiration: this.randomFutureDate(timeStamp),
+          type: 'INPUT_ACTION',
+          input_type: 'endpoint',
+          data: {
+            command: this.randomIsolateCommand(),
+            comment: this.randomString(15),
+          },
+        },
+        error: undefined,
+        user: {
+          id: 'elastic',
```

Comment:
if this is going into our index, then yeah use `randomUser()` -- since we will be writing into this as the logged in user. If this is going into the `.fleet-actions` (and I don't _think_ it is) then `elastic` is appropriate since we will continue writing into _there_ as system user.

Reply:
done d3a2b33b8404c37a21c538ae3dc24e5bd380ede5